boards/atxmega-a1u-xpro/Makefile.include

```diff
@@ -1,4 +1,7 @@
 AVRDUDE_PROGRAMMER = xplainedpro_pdi
 DEBUGPROTO = -X
 
+RAM_LEN = 512K
+LDSCRIPT_EXTRA = -Wl,--defsym=__heap_end=0x80FFFF
```

Comment:
I'm trying to find where `__heap_end` but I didn't come up with any results.

Reply:
If you are referring to docs, I look at [avr-libc malloc](https://www.nongnu.org/avr-libc/user-manual/malloc.html). 
The `__heap_end`  is appended at `makefiles/arch/avr8.inc.mk ` using `LINKFLAGS += $(LDSCRIPT_EXTRA)`.

Reply:
How is `__heap_end` calculated?
Is it internal RAM + external RAM?
Why is `RAM_LEN` < `__heap_end`?

Reply:
> How is __heap_end calculated?

Data memory for AVR-8 is defined at 0x80000 address for GCC. _The I/O memory, EEPROM, and SRAM will always have the same start addresses for all XMEGA devices. The address space for external memory will always start at the end of internal SRAM and end at address 0xFFFFFF. The internal SRAM always starts at hexadecimal address 0x2000. SRAM is accessed by the CPU using the load (LD/LDS/LDD) and store (ST/STS/STD) instructions._ In this case, RAMPZ/Y/X define the 24 bit address block. _The RAMPD is concatenated with the operand to enable direct addressing of the whole data memory space above 64KB. Together, RAMPD and the operand will form a 24-bit address._ Max SP addressable is 64k block and it is clear if there is a way to extend SP size.

To enable Max viable memory for RIOT-OS the heap_end should be 0x80000 + 64k (SP block) -1.

> Is it internal RAM + external RAM?

Yes, internal memory have precedence over external memory. This allows a contiguous space between internal + external memory. The base address should be defined `boards/<board>/include/periph_conf.h `. The memory align is mandatory based in the CS size. In this case, the CS will handle 256k external memory where first 64k is reserved for RIOT-OS. In fact, we can address the whole space but compiler not handle RAMPx registers. GCC assume they are always equal to Zero.

```
                      }, /* CS2 - SRAM        */
                      { EBI_CS_MODE_LPC_gc,
                        EBI_CS_ASPACE_256KB_gc,
                        EBI_CS_SRWS_1CLK_gc,
                        0x0UL,
                      },
```

> Why is RAM_LEN < __heap_end?

RAM_LEN should define MAX memory for RIOT-OS. That value should be 64k instead max memory installed. GCC will not touch that memory because assume RAMPx are always 0. User can access using the special methods that handle 24 bit pointer and return RAMPx register to 0 after their use.

Reply:
Note that only memory directly accessible is suitable for use via `malloc`, as `void *` pointers are 16 bit in size on AVR.